CLOUDSTACK-9184: [VMware] vmware.ports.per.dvportgroup global setting is not useful from vCenter 5.0 onwards

sureshanaparti opened the pull request:
Modify the default value of this setting, which is currently 256, by decreasing it to 8 because even a dvportgroup with autoexpand feature enabled there exists an initial number of ports as starting point, which can be again configured through this configuration setting. But keeping this value 256 doesn't make sense when auto expand feature is available. Keeping the starting number of dvports per dvportgroup to reasonably minimum would ensure optimal distribution across dvportgroups in dvSwitch.

The vSphere Auto Expand feature (introduced in vSphere 5.0) will take care of dynamically increasing/decreasing the dvPorts when running out of distributed ports . But in case of vSphere 4.1/4.0 (If used), as this feature is not there, the new default value (=> 8) have an impact in the existing deployments. Action item for vSphere 4.1/4.0: Admin should modify the global configuration setting "vmware.ports.per.dvportgroup" from 8 to any number based on their environment because the proposal default value of 8 would be very less without auto expand feature in general. The current default value of 256 may not need immediate modification after deployment, but 8 would be very less which means admin need to update immediately after upgrade.

GabrielBrascher commented:
@sureshanaparti Although it might be a bit tricky, I believe that the ideal solution would be to check for the hypervisor version; if it is 4.1/4.0 then the default value of 8 is changed to 256 (or any recommended amount for the given version). This way does not rely on admin to manually configure it (the idea would be to consider the global configuration if the admin set it, would just change a default value).

I checked and maybe a method for check the hypervisor version and set the **_portsPerDvPortGroup** could be called by **com.cloud.hypervisor.vmware.manager.VmwareManagerImpl.addHostToPodCluster(VmwareContext, long, Long, Long, String)**. This method would search a host using the clusterId (**com.cloud.host.dao.HostDaoImpl.findByClusterId(Long)**); with the given HostVO it verifies the hypervisor version (**com.cloud.host.HostVO.getHypervisorVersion()**) to set the default value according this version.

Thanks.

bvbharatk commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 161
 Hypervisor xenserver
 NetworkType Advanced
 Passed=110
 Failed=18
 Skipped=4

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vpc_router_nics.py

 * test_02_VPC_default_routes Failed

* test_vpc_redundant.py

 * test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL Failed

 * test_02_redundant_VPC_default_routes Failed

* test_privategw_acl.py

 * test_01_vpc_privategw_acl Failed

* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 2 runs

* test_router_dhcphosts.py

 * test_router_dhcphosts Failed

 * ContextSuite context=TestRouterDHCPHosts>:teardown Failed

* test_quota.py

 * test_01_quota Failed

 * test_02_quota Failed

 * test_03_quota Failed

 * test_04_quota Failed

 * test_05_quota Failed

 * test_06_quota Failed

 * test_07_quota Failed

* test_password_server.py

 * test_isolate_network_password_server Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_deploy_vgpu_enabled_vm
test_06_copy_template
test_06_copy_iso

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_guest_vlan_range.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_non_contigiousvlan.py
test_deploy_vm_iso.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_ssvm.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

rhtyd commented:
@sureshanaparti please rebase against latest master, move the schema changes to 481-490 upgrade path

tag:vmware-pickup

bvbharatk commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 71
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

sureshanaparti commented:
Addressed all the changes suggested and rebased. With vSphere 4.0/4.1, when the ports per dvportgroup is set to less than 256 (current default), then it is set to the default 256, else it takes the actual config value.

sureshanaparti commented:
Moved the schema changes to 4920-41000 upgrade path.

sureshanaparti commented:
@rhtyd  Can you please run trillian tests for this PR?

cloudmonger commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 396
 Hypervisor xenserver
 NetworkType Advanced
 Passed=103
 Failed=2
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_routers_network_ops.py

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

sateesh-chodapuneedi commented:
@sureshanaparti Can you please investigate the two failures listed in results from @cloudmonger?

sureshanaparti commented:
Please ignore the above BVT test results from @cloudmonger. These tests were executed on xenserver instead of VMware.

sureshanaparti commented:
@sateesh-chodapuneedi These two test failures are due to the ssh failure to the router VM, are not related to this PR changes.

_test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed_

> Failed to SSH into the Virtual Machine: SSH connection has Failed. Waited 150s. Error is SSH Connection Failed
> === TestName: test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | Status : FAILED ===
> "/automation/virtenv/00-16-3e-17-43-25/396/test/integration/smoke/test_routers_network_ops.py", line 338, in test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true\n    result = check_router_command(virtual_machine, nat_rule.ipaddress, ssh_command, check_string, self)\n', '  File "/automation/virtenv/00-16-3e-17-43-25/396/test/integration/smoke/test_routers_network_ops.py", line 64, in check_router_command\n    test_case.fail("Failed to SSH into the Virtual Machine: %s" % e)\n', '  File "/usr/local/lib/python2.7/unittest/case.py", line 412, in fail\n    raise self.failureException(msg)\n', 'AssertionError: Failed to SSH into the Virtual Machine: SSH connection has Failed. Waited 150s. Error is SSH Connection Failed\n']


_test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed_

> Failed to SSH into the Virtual Machine: SSH connection has Failed. Waited 150s. Error is SSH Connection Failed
> === TestName: test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | Status : FAILED ===
> 
> "/automation/virtenv/00-16-3e-17-43-25/396/test/integration/smoke/test_routers_network_ops.py", line 502, in test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false\\n    result = check_router_command(virtual_machine, nat_rule.ipaddress, ssh_command, check_string, self)\\n\', \'  File "/automation/virtenv/00-16-3e-17-43-25/396/test/integration/smoke/test_routers_network_ops.py", line 64, in check_router_command\\n    test_case.fail("Failed to SSH into the Virtual Machine: %s" % e)\\n\', \'  File "/usr/local/lib/python2.7/unittest/case.py", line 412, in fail\\n    raise self.failureException(msg)\\n\', \'AssertionError: Failed to SSH into the Virtual Machine: SSH connection has Failed. Waited 150s. Error is SSH Connection Failed\\n\']

sureshanaparti commented:
@rhtyd @borisroman can you kick off vmware tests?

rhtyd commented:
@sureshanaparti sure
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✖debian. JID-546

sureshanaparti commented:
@nvazquez @serg38 Please review the code changes.

serg38 commented:
@sureshanaparti Why don't we just remove Config for this setting and leave default to be 8? 
Your logic to handle v4.1 will take care of old installations.
As of May 2016 Vsphere 4.X is  not longer supported so we technically shouldn't even worry about it. 
@rhtyd @karuturi At some point we should remove vSphere 4.1 from a list of supported hypervisors.

sureshanaparti commented:
@rhtyd @karuturi Is this documented in any of the earlier release notes. Can we remove the config param "vmware.ports.per.dvportgroup" if vSphere 4.x is not supported?

sureshanaparti commented:
@rhtyd @karuturi Could see vSphere 4.1 in the latest installation guide. Please note that this documentation needs to be updated at some point later.
> vSphere and vCenter, versions 4.1, 5.0, 5.1 or 5.5.
> VMware vCenter Standard Edition 4.1, 5.0, 5.1 or 5.5 must be installed and available to manage the vSphere hosts.
> CloudStack requires VMware vSphere 4.1, 5.0, 5.1 or 5.5. VMware vSphere 4.0 is not supported.

sateesh-chodapuneedi commented:
>Can we remove the config param "vmware.ports.per.dvportgroup" now or hold it sometime?
@sureshanaparti Yes, we can remove this config param.

sureshanaparti commented:
@sateesh-chodapuneedi Thanks for confirming this. I'll update the code to remove the config param "vmware.ports.per.dvportgroup".

sureshanaparti commented:
@serg38 @sateesh-chodapuneedi Removed the config parameter "vmware.ports.per.dvportgroup".  Ports per dvportgroup is defaulted to 8 and for vSphere 4.x, it is 256. Please review.

sureshanaparti commented:
@borisstoyanov Updated the changes. Can you please kick off the tests here.

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-710

borisstoyanov commented:
@blueorangutan test centos7 vmware-55u3

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + vmware-55u3) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-1079)</b>
Environment: vmware-55u3 (x2), Advanced Networking with Mgmt server 7
Total time taken: 57652 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr1253-t1079-vmware-55u3.zip
Intermitten failure detected: /marvin/tests/smoke/test_affinity_groups_projects.py
Intermitten failure detected: /marvin/tests/smoke/test_deploy_vm_root_resize.py
Intermitten failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermitten failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermitten failure detected: /marvin/tests/smoke/test_volumes.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Test completed. 45 look ok, 7 have error(s)


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_vpc_site2site_vpn | `Failure` | 307.28 | test_vpc_vpn.py
test_01_create_volume | `Failure` | 189.40 | test_volumes.py
test_04_rvpc_privategw_static_routes | `Failure` | 247.76 | test_privategw_acl.py
test_02_vpc_privategw_static_routes | `Failure` | 182.10 | test_privategw_acl.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Failure` | 176.61 | test_internal_lb.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 524.79 | test_vpc_redundant.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 1107.32 | test_vpc_redundant.py
ContextSuite context=TestDeployVmRootSize>:setup | `Error` | 0.00 | test_deploy_vm_root_resize.py
test_01_vpc_remote_access_vpn | Success | 227.12 | test_vpc_vpn.py
test_01_redundant_vpc_site2site_vpn | Success | 759.30 | test_vpc_vpn.py
test_02_VPC_default_routes | Success | 374.67 | test_vpc_router_nics.py
test_01_VPC_nics_after_destroy | Success | 792.31 | test_vpc_router_nics.py
test_05_rvpc_multi_tiers | Success | 773.25 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | Success | 1620.45 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | Success | 714.75 | test_vpc_redundant.py
test_09_delete_detached_volume | Success | 26.06 | test_volumes.py
test_06_download_detached_volume | Success | 110.95 | test_volumes.py
test_05_detach_volume | Success | 100.28 | test_volumes.py
test_04_delete_attached_volume | Success | 20.30 | test_volumes.py
test_03_download_attached_volume | Success | 20.32 | test_volumes.py
test_02_attach_volume | Success | 59.83 | test_volumes.py
test_change_service_offering_for_vm_with_snapshots | Success | 593.08 | test_vm_snapshots.py
test_03_delete_vm_snapshots | Success | 280.27 | test_vm_snapshots.py
test_02_revert_vm_snapshots | Success | 229.46 | test_vm_snapshots.py
test_01_create_vm_snapshots | Success | 161.70 | test_vm_snapshots.py
test_deploy_vm_multiple | Success | 343.70 | test_vm_life_cycle.py
test_deploy_vm | Success | 0.03 | test_vm_life_cycle.py
test_advZoneVirtualRouter | Success | 0.03 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | Success | 26.96 | test_vm_life_cycle.py
test_09_expunge_vm | Success | 125.24 | test_vm_life_cycle.py
test_08_migrate_vm | Success | 66.29 | test_vm_life_cycle.py
test_07_restore_vm | Success | 0.12 | test_vm_life_cycle.py
test_06_destroy_vm | Success | 10.17 | test_vm_life_cycle.py
test_03_reboot_vm | Success | 5.15 | test_vm_life_cycle.py
test_02_start_vm | Success | 25.27 | test_vm_life_cycle.py
test_01_stop_vm | Success | 10.16 | test_vm_life_cycle.py
test_CreateTemplateWithDuplicateName | Success | 402.76 | test_templates.py
test_08_list_system_templates | Success | 0.03 | test_templates.py
test_07_list_public_templates | Success | 0.04 | test_templates.py
test_05_template_permissions | Success | 0.06 | test_templates.py
test_04_extract_template | Success | 20.28 | test_templates.py
test_03_delete_template | Success | 5.12 | test_templates.py
test_02_edit_template | Success | 90.18 | test_templates.py
test_01_create_template | Success | 201.40 | test_templates.py
test_10_destroy_cpvm | Success | 297.07 | test_ssvm.py
test_09_destroy_ssvm | Success | 269.05 | test_ssvm.py
test_08_reboot_cpvm | Success | 186.85 | test_ssvm.py
test_07_reboot_ssvm | Success | 158.95 | test_ssvm.py
test_06_stop_cpvm | Success | 206.99 | test_ssvm.py
test_05_stop_ssvm | Success | 239.37 | test_ssvm.py
test_04_cpvm_internals | Success | 1.26 | test_ssvm.py
test_03_ssvm_internals | Success | 3.79 | test_ssvm.py
test_02_list_cpvm_vm | Success | 0.14 | test_ssvm.py
test_01_list_sec_storage_vm | Success | 0.15 | test_ssvm.py
test_02_list_snapshots_with_removed_data_store | Success | 258.32 | test_snapshots.py
test_01_snapshot_root_disk | Success | 36.40 | test_snapshots.py
test_04_change_offering_small | Success | 97.65 | test_service_offerings.py
test_03_delete_service_offering | Success | 0.04 | test_service_offerings.py
test_02_edit_service_offering | Success | 0.08 | test_service_offerings.py
test_01_create_service_offering | Success | 0.34 | test_service_offerings.py
test_02_sys_template_ready | Success | 0.21 | test_secondary_storage.py
test_01_sys_vm_start | Success | 0.22 | test_secondary_storage.py
test_09_reboot_router | Success | 181.24 | test_routers.py
test_08_start_router | Success | 156.07 | test_routers.py
test_07_stop_router | Success | 25.26 | test_routers.py
test_06_router_advanced | Success | 0.07 | test_routers.py
test_05_router_basic | Success | 0.04 | test_routers.py
test_04_restart_network_wo_cleanup | Success | 5.94 | test_routers.py
test_03_restart_network_cleanup | Success | 171.21 | test_routers.py
test_02_router_internal_adv | Success | 1.05 | test_routers.py
test_01_router_internal_basic | Success | 0.57 | test_routers.py
test_router_dns_guestipquery | Success | 77.56 | test_router_dns.py
test_router_dns_externalipquery | Success | 0.09 | test_router_dns.py
test_router_dhcphosts | Success | 199.89 | test_router_dhcphosts.py
test_router_dhcp_opts | Success | 21.82 | test_router_dhcphosts.py
test_01_updatevolumedetail | Success | 5.14 | test_resource_detail.py
test_01_reset_vm_on_reboot | Success | 95.94 | test_reset_vm_on_reboot.py
test_createRegion | Success | 0.04 | test_regions.py
test_create_pvlan_network | Success | 5.24 | test_pvlan.py
test_dedicatePublicIpRange | Success | 0.45 | test_public_ip_range.py
test_03_vpc_privategw_restart_vpc_cleanup | Success | 1356.50 | test_privategw_acl.py
test_01_vpc_privategw_acl | Success | 228.38 | test_privategw_acl.py
test_03_migration_options_storage_tags | Success | 37.12 | test_primary_storage.py
test_02_edit_primary_storage_tags | Success | 0.13 | test_primary_storage.py
test_01_primary_storage_nfs | Success | 38.53 | test_primary_storage.py
test_01_deploy_vms_storage_tags | Success | 46.57 | test_primary_storage.py
test_createPortablePublicIPRange | Success | 15.22 | test_portable_publicip.py
test_createPortablePublicIPAcquire | Success | 15.48 | test_portable_publicip.py
test_isolate_network_password_server | Success | 93.47 | test_password_server.py
test_UpdateStorageOverProvisioningFactor | Success | 0.16 | test_over_provisioning.py
test_oobm_zchange_password | Success | 30.97 | test_outofbandmanagement.py
test_oobm_multiple_mgmt_server_ownership | Success | 16.38 | test_outofbandmanagement.py
test_oobm_issue_power_status | Success | 10.26 | test_outofbandmanagement.py
test_oobm_issue_power_soft | Success | 15.48 | test_outofbandmanagement.py
test_oobm_issue_power_reset | Success | 15.47 | test_outofbandmanagement.py
test_oobm_issue_power_on | Success | 10.41 | test_outofbandmanagement.py
test_oobm_issue_power_off | Success | 15.38 | test_outofbandmanagement.py
test_oobm_issue_power_cycle | Success | 10.33 | test_outofbandmanagement.py
test_oobm_enabledisable_across_clusterzones | Success | 87.87 | test_outofbandmanagement.py
test_oobm_enable_feature_valid | Success | 5.18 | test_outofbandmanagement.py
test_oobm_enable_feature_invalid | Success | 0.17 | test_outofbandmanagement.py
test_oobm_disable_feature_valid | Success | 5.20 | test_outofbandmanagement.py
test_oobm_disable_feature_invalid | Success | 0.12 | test_outofbandmanagement.py
test_oobm_configure_invalid_driver | Success | 0.09 | test_outofbandmanagement.py
test_oobm_configure_default_driver | Success | 0.10 | test_outofbandmanagement.py
test_oobm_background_powerstate_sync | Success | 18.55 | test_outofbandmanagement.py
test_extendPhysicalNetworkVlan | Success | 15.44 | test_non_contigiousvlan.py
test_01_nic | Success | 606.61 | test_nic.py
test_releaseIP | Success | 308.67 | test_network.py
test_reboot_router | Success | 640.20 | test_network.py
test_public_ip_user_account | Success | 10.34 | test_network.py
test_public_ip_admin_account | Success | 40.34 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 77.15 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.92 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 155.19 | test_network.py
test_delete_account | Success | 308.46 | test_network.py
test_02_port_fwd_on_non_src_nat | Success | 55.72 | test_network.py
test_01_port_fwd_on_src_nat | Success | 111.89 | test_network.py
test_nested_virtualization_vmware | Success | 490.01 | test_nested_virtualization.py
test_nic_secondaryip_add_remove | Success | 303.59 | test_multipleips_per_nic.py
test_list_zones_metrics | Success | 0.30 | test_metrics_api.py
test_list_volumes_metrics | Success | 5.56 | test_metrics_api.py
test_list_vms_metrics | Success | 272.52 | test_metrics_api.py
test_list_pstorage_metrics | Success | 0.34 | test_metrics_api.py
test_list_infrastructure_metrics | Success | 0.42 | test_metrics_api.py
test_list_hosts_metrics | Success | 0.44 | test_metrics_api.py
test_list_clusters_metrics | Success | 0.42 | test_metrics_api.py
login_test_saml_user | Success | 19.43 | test_login.py
test_assign_and_removal_lb | Success | 150.25 | test_loadbalance.py
test_02_create_lb_rule_non_nat | Success | 208.42 | test_loadbalance.py
test_01_create_lb_rule_src_nat | Success | 210.00 | test_loadbalance.py
test_03_list_snapshots | Success | 0.14 | test_list_ids_parameter.py
test_02_list_templates | Success | 0.05 | test_list_ids_parameter.py
test_01_list_volumes | Success | 0.03 | test_list_ids_parameter.py
test_07_list_default_iso | Success | 0.08 | test_iso.py
test_05_iso_permissions | Success | 0.07 | test_iso.py
test_04_extract_Iso | Success | 5.17 | test_iso.py
test_03_delete_iso | Success | 95.22 | test_iso.py
test_02_edit_iso | Success | 0.11 | test_iso.py
test_01_create_iso | Success | 21.11 | test_iso.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Success | 643.32 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Success | 1175.57 | test_internal_lb.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Success | 954.49 | test_internal_lb.py
test_dedicateGuestVlanRange | Success | 10.46 | test_guest_vlan_range.py
test_UpdateConfigParamWithScope | Success | 0.16 | test_global_settings.py
test_rolepermission_lifecycle_update | Success | 6.27 | test_dynamicroles.py
test_rolepermission_lifecycle_list | Success | 6.05 | test_dynamicroles.py
test_rolepermission_lifecycle_delete | Success | 5.93 | test_dynamicroles.py
test_rolepermission_lifecycle_create | Success | 6.11 | test_dynamicroles.py
test_rolepermission_lifecycle_concurrent_updates | Success | 6.12 | test_dynamicroles.py
test_role_lifecycle_update_role_inuse | Success | 6.07 | test_dynamicroles.py
test_role_lifecycle_update | Success | 6.11 | test_dynamicroles.py
test_role_lifecycle_list | Success | 6.11 | test_dynamicroles.py
test_role_lifecycle_delete | Success | 11.14 | test_dynamicroles.py
test_role_lifecycle_create | Success | 6.14 | test_dynamicroles.py
test_role_inuse_deletion | Success | 6.07 | test_dynamicroles.py
test_role_account_acls_multiple_mgmt_servers | Success | 8.68 | test_dynamicroles.py
test_role_account_acls | Success | 9.13 | test_dynamicroles.py
test_default_role_deletion | Success | 6.43 | test_dynamicroles.py
test_04_create_fat_type_disk_offering | Success | 0.10 | test_disk_offerings.py
test_03_delete_disk_offering | Success | 0.08 | test_disk_offerings.py
test_02_edit_disk_offering | Success | 0.07 | test_disk_offerings.py
test_02_create_sparse_type_disk_offering | Success | 0.13 | test_disk_offerings.py
test_01_create_disk_offering | Success | 0.23 | test_disk_offerings.py
test_deployvm_userdispersing | Success | 136.63 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userconcentrated | Success | 66.28 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_firstfit | Success | 277.51 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userdata_post | Success | 81.18 | test_deploy_vm_with_userdata.py
test_deployvm_userdata | Success | 222.06 | test_deploy_vm_with_userdata.py
test_deploy_vm_from_iso | Success | 324.12 | test_deploy_vm_iso.py
test_3d_gpu_support | Success | 597.36 | test_deploy_vgpu_enabled_vm.py
test_DeployVmAntiAffinityGroup | Success | 428.71 | test_affinity_groups.py
test_08_resize_volume | Skipped | 5.12 | test_volumes.py
test_07_resize_fail | Skipped | 10.33 | test_volumes.py
test_06_copy_template | Skipped | 0.00 | test_templates.py
test_static_role_account_acls | Skipped | 0.04 | test_staticroles.py
test_11_ss_nfs_version_on_ssvm | Skipped | 0.03 | test_ssvm.py
test_01_scale_vm | Skipped | 66.47 | test_scale_vm.py
test_01_primary_storage_iscsi | Skipped | 0.05 | test_primary_storage.py
test_06_copy_iso | Skipped | 0.00 | test_iso.py
test_06_verify_guest_lspci_again | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_05_change_vm_ostype_restart | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_04_verify_guest_lspci | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_03_verify_libvirt_attach_disk | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_02_verify_libvirt_after_restart | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_01_verify_libvirt | Skipped | 0.00 | test_deploy_virtio_scsi_vm.py
test_deploy_vgpu_enabled_vm | Skipped | 1.20 | test_deploy_vgpu_enabled_vm.py

cloudmonger commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 696
 Hypervisor xenserver
 NetworkType Advanced
 Passed=112
 Failed=0
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

cloudmonger commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 777
 Hypervisor xenserver
 NetworkType Advanced
 Passed=110
 Failed=2
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed

* test_volumes.py

 * test_06_download_detached_volume Failing since 2 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

rhtyd commented:
@sureshanaparti ping, can you fix conflicts?

PaulAngus commented:
There is an issue with existing environments, which may be using dvsportgroups.  If portgroup has more than (say) 8 ports in use already then a error such as:
`Cannot set number of ports to 10 because at least 11 out of 18 ports are still in use`
will be thrown when creating a VM on said switch and the VM will fail to be deployed.
Setting ALL dvsportgroups to use a sufficiently high number will be expensive in large environments.

Code needs to set number of ports to slightly higher than the number of ports in use, on a per  portgroup basis.

rhtyd commented:
Closing this, will send a new PR taking over this issue.

sureshanaparti commented:
@PaulAngus , @rhtyd  I'll continue work on the changes required if any, on this.

rhtyd commented:
@sureshanaparti I've moved the changes to this new PR against 4.11: https://github.com/apache/cloudstack/pull/2634#issuecomment-387994777
Perhaps you can share how you tested it, help with testing in general.
